spec-tests/src/test/java/org/candlepin/spec/PoolSpecTest.java

```diff
@@ -0,0 +1,673 @@
+    }
+
+    // ugh...
+    private AttributeDTO buildAttribute(String key, String value) {
+        return new AttributeDTO()
+            .name(key)
+            .value(value);
+    }
+
+    private PoolDTO createPool(ApiClient client, OwnerDTO owner) throws ApiException {
+        ProductDTO product = this.createProduct(client, owner);
+        return this.createPool(client, owner, product);
+    }
+
+    private PoolDTO createPool(ApiClient client, OwnerDTO owner, ProductDTO product) throws ApiException {
+        OffsetDateTime startDate = OffsetDateTime.now();
+        OffsetDateTime endDate = startDate.plusYears(1);
+
+        PoolDTO pool = new PoolDTO()
+            .productId(product.getId())
+            .quantity(10L)
+            .startDate(startDate)
+            .endDate(endDate)
+            .subscriptionId(this.randomSuffix("source_sub"))
+            .subscriptionSubKey("master")
+            .upstreamPoolId(this.randomSuffix("upstream_pool_id"));
+
+        return client.owners()
+            .createPool(owner.getKey(), pool);
+    }
+
+    @Test
+    public void letsConsumersViewPools() throws ApiException {
+        ApiClient adminClient = ApiClients.admin();
+        OwnerDTO owner = this.createOwner(adminClient);
+        PoolDTO pool = this.createPool(adminClient, owner);
+
+        ApiClient userClient = this.createUserClient(adminClient, owner);
+
+        List<PoolDTO> pools = userClient.pools()
+            .listPools(owner.getId(), null, null, true, null);
+
+        assertNotNull(pools);
+        assertEquals(1, pools.size());
+
+        ApiClient consumerClient = this.createConsumerClient(userClient, owner);
+
+        PoolDTO fetched = consumerClient.pools()
+            .getPool(pool.getId(), null, null);
+
+        assertNotNull(fetched);
+        assertEquals(pool.getId(), fetched.getId());
+    }
+
+  // it 'includes null values in json' do
+  //   owner1 = create_owner random_string('test_owner')
+  //   owner1_client = user_client(owner1, random_string('testuser'))
+
+  //   product = create_product(nil, nil, :owner => owner1['key'])
+  //   create_pool_and_subscription(owner1['key'], product.id, 10)
+  //   @cp.refresh_pools(owner1['key'])
+  //   pool = owner1_client.list_pools(:owner => owner1.id).first
+  //   pool.member?("upstreamPoolId").should == true
+  //   pool["upstreamPoollId"].should be_nil
+  //   pool.member?("upstreamEntitlementId").should == true
+  //   pool["upstreamEntitlementId"].should be_nil
+  //   pool.member?("upstreamConsumerId").should == true
+  //   pool["upstreamConsumerId"].should be_nil
+  // end
+
+    // @Test
+    // public void includesNullValuesInJSON() {
+        // This test isn't possible given that we don't know if the fields in question were present in the
+        // JSON by the time we can operate on the data via DTO.
+    // }
+
+    @Test
+    public void doesNotLetConsumersViewPoolEntitlements() throws ApiException {
```

Comment:
Can we use the `@DisplayName("")` annotation to specify human readable names for each test case?

Reply:
I guess? Have we standardized on any verbiage, tense, or such for the names? If we're going to switch to something that makes quickly jumping to a test worse, I hope we're at least making text consistent.

Reply:
Done. Since we're (to my knowledge) lacking a convention, I'm standardizing on "should" and "should not" without a leading noun, and reasonable descriptions of what the test does from there. Additionally, I updated the test names to match the display name so we can still find tests by quickly looking up the symbol.

Reply:
ack